atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/AtlasBackupService.java

```diff
@@ -109,16 +141,44 @@ public static AtlasBackupService createForTests(
     private void storeBackupToken(InProgressBackupToken backupToken) {
         inProgressBackups.put(backupToken.getNamespace(), backupToken);
         backupPersister.storeImmutableTimestamp(backupToken);
+        lockRefresher.registerLocks(ImmutableSet.of(backupToken));
     }
 
     public Set<Namespace> completeBackup(Set<Namespace> namespaces) {
         Set<InProgressBackupToken> tokens = namespaces.stream()
                 .map(inProgressBackups::remove)
                 .filter(Objects::nonNull)
                 .collect(Collectors.toSet());
+        lockRefresher.unregisterLocks(tokens);
+
+        Set<Namespace> namespacesWithInProgressBackups =
+                tokens.stream().map(InProgressBackupToken::getNamespace).collect(Collectors.toSet());
+        if (tokens.size() < namespaces.size()) {
+            Set<Namespace> namespacesWithNoInProgressBackup =
+                    Sets.difference(namespaces, namespacesWithInProgressBackups);
```

Comment:
i would say make a copy of the setView of diff unless namespaces and namespacesWithInProgressBackups are both immutable

Reply:
do we need to copy it? the javadoc of `Sets.difference` states that the view is unmodifiable.